src/main/java/codeu/controller/NewPasswordServlet.java

```diff
@@ -0,0 +1,59 @@
+package codeu.controller;
+
+import codeu.model.data.User;
+import codeu.model.store.basic.UserStore;
+import org.mindrot.jbcrypt.BCrypt;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class NewPasswordServlet extends HttpServlet {
+    /** Store class that gives access to Users. */
+    private UserStore userStore;
+
+    @Override
+    public void init() throws ServletException {
+        super.init();
+        setUserStore(UserStore.getInstance());
+    }
+
+    /**
+     * Sets the UserStore used by this servlet. This function provides a common setup method for use
+     * by the test framework or the servlet's init() function.
+     */
+    void setUserStore(UserStore userStore) {
+        this.userStore = userStore;
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        request.getRequestDispatcher("/WEB-INF/view/newPassword.jsp").forward(request, response);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        String newPassword = request.getParameter("newPassword");
+        String reTypedNewPassword = request.getParameter("reTypedNewPassword");
+        String username =request.getParameter("usernameVerification");
+
```

Comment:
Remember to try and clean the user inputs from XSS and related attacks! Check the Jsoup library and methods we used at `ProfileServlet.java` and `AdminServlet.java` (doPost section). What it does is takes out html tags, etc!

Reply:
okay will do, thanks!

Reply:
okay I checked them out, what does clean the user inputs mean and what do the Jsoup methods actually do?

Reply:
They take out html tags, if we don't do this a malicious user could input <script> doSomethingMalicious <script> and inject code into our server!

Reply:
Okay I added the Jsoup method, is this correct? also shouldn't that be added to RegisterServlet.Java too?

Reply:
Yeah, if I'm not mistaken the first if filters those tags and special characters using regex (short for regular expressions). :)